webapp/server.js

```diff
@@ -83,8 +98,19 @@ function validateRequest(req) {
         return { status: 400, error: "Please enter a URL so the crawler crawler can show you how cool it is."};
     }
 
+    if (!isValidUrl(req.body.url)) {
+        return { status: 400, error: "Url invalid."};
+    }
+
     if (req.body.depth == null) {
         return { status: 400, error: "A depth is required to begin crawling. Please enter a depth."};
+
+    }
+
+    if(req.body.keyword != null ) {
+        //check only one word and
+        if(req.body.keyword.length <= 2)
```

Comment:
This is so good! Let me add a function to React Component to display the error message to the user.

Reply:
I should probably collapse that into one single statement. 
much thanks!!